client/dynamicplugins/registry_test.go

```diff
@@ -47,51 +57,58 @@ func TestPluginEventBroadcaster_UnsubscribeWorks(t *testing.T) {
 
 	ch1 := b.subscribe()
 
-	listenFunc := func(ch chan *PluginUpdateEvent, updateBool *bool) {
-		select {
-		case e := <-ch:
-			if e == nil {
-				*updateBool = true
+	var wg sync.WaitGroup
+	wg.Add(1)
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	defer cancel()
+
+	go func() {
+		defer wg.Done()
+		for {
+			select {
+			case <-ctx.Done():
+				t.Fail()
+				return
+			case <-ch1:
+				rcv1 = true
+				if rcv1 {
+					return
+				}
```

Comment:
`rcv1` seems like a noop. Can this be replaced with `return`?

Reply:
Oops, yup. Copypasta from the previous test. Will fix.